### PR TITLE
OCPQE-17433: Add OperatorLifecycleManager Cap for v415

### PR DESCRIPTION
**Pull request by yunjiang29**
https://github.com/openshift/installer/pull/7495
https://github.com/openshift/api/pull/1589
https://github.com/openshift/api/pull/1629

failed CI job: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-baselinecaps-none-custom-sg-f14/1714915210466168832

### CODE REVIEW

**Comment by yunjiang29** [comment]
cc @jinyunma @jianlinliu for awareness of this change, still in WIP

**Comment by jianlinliu** [comment]
> https://github.com/openshift/installer/pull/7495

I am not sure if this will bring some change for `enable-qe-catalogsource` step, but at least from the [job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-baselinecaps-none-custom-sg-f14/1714915210466168832/artifacts/aws-ipi-baselinecaps-none-custom-sg-f14/enable-qe-catalogsource/build-log.txt), even we set `none` baseline capacity, sounds like the step get passed. cc @jianzhangbjz

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 1]
```diff
@@ -84,6 +84,7 @@ caps_operator[Storage]="storage"
 caps_operator[NodeTuning]="node-tuning"
 caps_operator[MachineAPI]="machine-api control-plane-machine-set cluster-autoscaler"
 caps_operator[ImageRegistry]="image-registry"
+caps_operator[OperatorLifecycleManager]="operator-lifecycle-manager"
```
Comment (jinyunma) [inline ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:87]: this capability includes several operators:
```
caps_operator[OperatorLifecycleManager]="operator-lifecycle-manager operator-lifecycle-manager-catalog operator-lifecycle-manager-packageserver"
```

**Comment by yunjiang29** [comment]
> Since v4.15 cap set is not supported on installer and API, dev is working on it ([openshift/api#1629](https://github.com/openshift/api/pull/1629)), so I'm waiting v4.15 cap set be added and then fix all related steps in Prow.

Ha, the PR just got merged 20 hours ago.

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh]: Updated.

**Comment by jianlinliu** [comment]
> but still need to bump this API version on installer and CVO, then v4.15 cap set should be worked.

We just need to wait for installer and CVO get bumped to check the future ci jobs results, but no need to update the script any more once this PR get merged, right?

**Comment by jinyunma** [comment]
> > but still need to bump this API version on installer and CVO, then v4.15 cap set should be worked.
> 
> We just need to wait for installer and CVO get bumped to check the future ci jobs results, but no need to update the script any more once this PR get merged, should not block the PR merging, right?

From installer PR, I also found the pre-check for marketplace and OperatorLifecycleManage.
`the marketplace capability requires the OperatorLifecycleManage`

when `OperatorLifecycleManage` is selected to be disabled, should disable `marketplace` together.

so also need to add similar logical in `ipi-conf-capability-additonalcaps` as what we [did for ImageRegistry and Build](https://github.com/openshift/release/pull/44644/files#diff-50fb3aa752ba71ff39d5d0ffc98d23226b37e25553b6c6d225dc5e57c9636aecL93-L97).

**Comment by jianlinliu** [comment]
> when OperatorLifecycleManage is selected to be disabled, should disable marketplace together.

Ah, right.

**Comment by jinyunma** [comment]
@yunjiang29 step [ipi-conf-manual-creds-remove-unnecessary-creds](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh) also needs to be update to support v4.15.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-baselinecaps-none-custom-sg-f14

**Comment by yunjiang29** [comment]
The change is available in `4.15.0-0.nightly-2023-10-20-214244`, but CI job picks up `4.15.0-0.nightly-2023-10-17-065657` ATM, so the [rehearse job failed](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/44644/rehearse-44644-periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-baselinecaps-none-custom-sg-f14/1716352079842250752).

**Comment by yunjiang29** [comment]
@jianlinliu @jinyunma do you think we can merge this right now or wait for a build available to test.

**Comment by jianlinliu** [comment]
/lgtm

leave @jinyunma for final confirm

**Comment by jianlinliu** [comment]
From the rehearse job log:
```
ERROR: capability OperatorLifecycleManager: operator operator-lifecycle-manager -- found
ERROR: capability OperatorLifecycleManager: operator operator-lifecycle-manager-catalog -- found
ERROR: capability OperatorLifecycleManager: operator operator-lifecycle-manager-packageserver -- found
```

Is that a known issue?

**Comment by jinyunma** [comment]
> From the rehearse job log:
> 
> ```
> ERROR: capability OperatorLifecycleManager: operator operator-lifecycle-manager -- found
> ERROR: capability OperatorLifecycleManager: operator operator-lifecycle-manager-catalog -- found
> ERROR: capability OperatorLifecycleManager: operator operator-lifecycle-manager-packageserver -- found
> ```
> 
> Is that a known issue?

because https://github.com/openshift/installer/pull/7495 is not landed into accepted nightly build.

**Pull request merged**